lib/cocoapods/generator/copy_resources_script.rb

```diff
@@ -71,6 +82,28 @@ def install_resources_function
         end
       end
 
+      # @return [String] The contents of the target xcassets script.
+      #
+      def target_xcassets
+
+        project_path = user_project_path.to_path
+        user_target_name = target_name.sub("Pods-", "")
+
+        script = <<-EOS 
+        require 'xcodeproj';
+        project_path = '#{project_path}';
+        target_name = '#{user_target_name}';
+        project = Xcodeproj::Project.open(project_path);
```

Comment:
this also has slowdown implications for large projects. Each target script phase will actually re-open the xcodeproj to calculate these which can be fairly slow.

We have to come up with a different strategy to figure this out. I'd like to understand what was the reason the initial script phase was so naive.

Reply:
For example this is very old code (back in 2013 and 2015) as this commit https://github.com/CocoaPods/CocoaPods/commit/e06296847aa99d0e31cee1451dab9995200ec4b5

Reply:
Another approach could be adding the pod ‘.xcassets’ in the user project during ‘pod install’. That way the assets catalogue doesn’t need to be recompiled.

However, this could complicate ‘pod deintegrate’

Reply:
There is also `${PODS_TARGET_SRCROOT}` now set which is always relative to where the pod is (local dev pod or remote). 

I am playing around with a few samples to try to understand existing logic.

Reply:
If I understood it correctly, this is what’s happening:
1. The user project compiles ‘.xcassets’ into Assets.car
2. There is a pod shipped as a static library (scenario I’m experiencing)
3. The pod expects the assets to be in the the main bundle
4. Copy resources copies all resources into the main bundle
5. Since ‘.xcassets’ need to be compiled, and adding to Assets.car is impossible, it is regenerated
6. The script picks up all ‘.xcassets’ to compile them, accidentally getting some that are not intended to go in

Reply:
@i-dama you are correct. Still trying to understand what was the reasoning of adding 
```sh
  # Find all other xcassets (this unfortunately includes those of path pods and other targets).
  OTHER_XCASSETS=$(find "$PWD" -iname "*.xcassets" -type d)
```

Reply:
@i-dama I am understanding this more. It seems its trying to pick up the xcassets used by the app so it can compile all of them together.

Reply:
What about the approach suggested above? Adding the pod .xcassets in the app’s “copy bundle resources” phase during ‘pod install’ and avoiding recompiling the Assets.car altogether

Reply:
it would probably work, but then de integrate will become a problem.

Reply:
It would also pollute the users project further.

Reply:
@i-dama the only way I can think of doing this, is what you mentioned with polluting the users app with these references.

Reply:
@i-dama admittedly another crappy way to resolve this provide an ENV variable that is set by the user that specifies the xcconfigs that part of their app. Its pretty crappy but I cant think of anything better without literally re-designing how all of the resource copying works in CocoaPods.

Reply:
Agree. I think any solution that requires the user to have to manually do something whenever they add an `.xcassets` would be requiring too much from them.

I'm going to do a bit more digging, and try to come up with another proposal. I think people might be affected by this without realizing it, since you wouldn't notice the extra images in the `.ipa` file unless you manually inspect it

Anyway, thanks for taking the time to investigate this 💯

Reply:
Given where we are this might work. So you will be able to find which `.xcassets` folder the `Name` exists into and use that?

Yes there could be some unwanted ones but much much smaller chance

Reply:
@i-dama it might be a bit scary too because if there are collisions we must ensure we dont pick the wrong one as it could be a different image

Reply:
@dnkoutso I'd propose this logic:
An `.xcassets` is included if *all* of the `xcimageset`s are present in the compiled `Assets.car` file.

From what I know, there is no way of instructing Xcode to compile only certain `.xcimageset`s in a given `.xcassets` catalogue, so by checking that *all* are present, we're most likely not going to exclude any `.xcassets` that should be included.

However, the problem would still persist if:
* Target 1 - Assets1: `1`, `2`, `3`
* Target 2 - Assets2: `1`, `2`

If compiling for `Target 1`, Assets2 would be included.

I don't see a way of knowing the distinction without tapping into the information from `.xcodeproj`.

However, the current state has the same problem, and doesn't handle collisions, so doing this would (hopefully) solve it for a subset of affected configurations.

Reply:
OK. want to try to make a PR for it? We *might* want to consider hiding it behind some sort of flag or ENV variable presence to ensure we slowly migrate to that.

Reply:
Seems like it will be a lot of bash scripting....

Reply:
Another approach would be to create a temporary `Pods.xcassets` catalogue, and copy all matching `.xcimageset`s into it.
After `Assets.car` is compiled, it can be deleted.

However, the collision problem still persists.

Reply:
> However, the collision problem still persists. you mean your initial (1,2,3) one but not the wrong image right? 

Could work as well. We need to ensure also the script phase does not run again (input and output paths) on subsequent builds.

Reply:
I don't see the distinction between the (1,2,3) and the wrong image problem.
In the 1,2,3 problem, images 1 and 2 get copied twice, and it's not known which one is going to get used.

I'd be glad to try and create a PR with it. Any preference on the approach? (I prefer the `Pods.xcassets` approach, seems safer)

Regarding the input/output, I don't think anything needs to be changed. The only thing changing is the logic of compiling `Assets.car` (Include all vs filter out some), the inputs/outputs stay the same.

Reply:
We could output an Xcode warning in that case.

You can go with `Pods.xcassets` since its all internal implementation detail of pods but I still believe we should hide this behind an ENV flag just to be sure we do not "break the world" (although admittedly is already broken ;))

Reply:
Actually, an input would be the `Assets.car` file. Probably not there atm.

Reply:
Alright 👍 
I'm going to try and make a PR this weekend. Should I just update this one or create a new one?
Agree on an ENV flag too 👍 

Funny thing is, `actool` (I believe) emits a warning for this, and it's also present in the current state. But it doesn't propagate to Xcode runtime warnings:
![image](https://user-images.githubusercontent.com/2226475/40554215-489b09d8-6045-11e8-8c33-9ad8fad16827.png)

Reply:
Let's make a new PR! We've had plenty of technical conversation here :)